### DISCUSSION TITLE
Input.http change value

### DISCUSSION THREAD

**Discussion by lowepalma** [savonet/liquidsoap#3065] [User Support]
I am using input.httml("https:examples.com:8000/stream) and then i want another connection be input.httml("https:examples222.com:8000/stream) i can't get it to change the url it always plays the same as it loads . Is there something to reset it to stop it, how would it be done

**Comment by vitoyucepi** [comment]
Hi @lowepalma,

```ruby
input_1 = input.http("https:examples.com:8000/stream")
input_2 = input.http("https:examples222.com:8000/stream")
```
[`input.http`](https://www.liquidsoap.info/doc-2.1.4/reference.html#input.http) will be available as long as the source is, due to its nature.
It's up to you to find a way to decide which input should be active.

There are multiple ways to resolve your problem.
1. Time-based schedule and the [`switch`](https://www.liquidsoap.info/doc-2.1.4/reference.html#switch).
   ```ruby
   radio = switch([
     ({8h-19h}, input_1),
     ({19h-8h}, input_2)
   ], track_sensitive=false)
   ```
   [`{8h-19h}`](https://www.liquidsoap.info/doc-2.1.4/language.html#time-predicates) is a [function](https://www.liquidsoap.info/doc-2.1.4/language.html#anonymous-function-with-no-arguments) that will become `true` if the local time is between `08:00` and `19:00`.
   You have the option to utilize more advanced functions here.
2. External variable and the [`switch`](https://www.liquidsoap.info/doc-2.1.4/reference.html#switch).
   ```ruby
   activate_input_1 = ref(false)
   activate_input_2 = ref(false)
   radio = switch([
     (!activate_input_1, input_1),
     (!activate_input_2, input_2)
   ], track_sensitive=false)
   ```
   `activate_input_1` and `activate_input_2` are set elsewhere.
   For instance, you can utilize [`telnet`](https://www.liquidsoap.info/doc-2.1.4/server.html#using-telnet), [`harbor.http`](https://www.liquidsoap.info/doc-2.1.4/harbor_http.html#harbor-as-http-server), [`interactive`](https://www.liquidsoap.info/doc-2.1.4/server.html#interactive-variables), or [`thread`](https://www.liquidsoap.info/doc-2.1.4/reference.html#thread.run).

**Reply by lowepalma** [reply]
How change value input_1 value from function ? the problem I have is that the value does not change, always if it has the first value.

EX: 

input_1 = ref(input.http("https://aaaaaaaaa:2957/stream"))

def one_new_value =
      input_1 = ref(input.http("https://bbbbbbb:2957/stream"))
end

The not change to https://bbbbbbb:2957/stream

**Reply by toots** [reply]
Hi @lowepalma ,

You need to set the variable outside of the source. Something like this:

```liquidsoap
url = ref("https://aaaaaaaaa:2957/stream")

s = input.http({!url}) # {!url} is the same as: fun () -> !url, i.e. a function that returns the latest value for url

# Then later:
url := "https://bbbbbbb:2957/stream"
```

If you are looking for a step-by-step explanation of this and more, I'd recommend the liquidsoap book: https://www.liquidsoap.info/doc-dev/book.html

**Reply by lowepalma** [reply]
http.input() has any stop, reset command?
